src/Symfony/Bridge/Twig/Extension/TranslationExtension.php

```diff
@@ -49,11 +49,19 @@ public function __construct($translator = null, NodeVisitorInterface $translatio
     }
 
     /**
-     * @deprecated since Symfony 4.2
+     * @return TranslatorInterface|null
      */
     public function getTranslator()
     {
-        @trigger_error(sprintf('The "%s()" method is deprecated since Symfony 4.2.', __METHOD__), E_USER_DEPRECATED);
+        if (null === $this->translator) {
+            if (!interface_exists(TranslatorInterface::class)) {
+                throw new \LogicException(sprintf('You cannot use the "%s" if the Translation Contracts are not available. Try running "composer require symfony/translation-contracts".', __CLASS__));
```

Comment:
What about recommending `symfony/translator` instead? If there are some usage of the Translation extension, it means that people want to use translation. So, they need the concrete implementation, not just the contracts.

Reply:
Why not. What's the use case for using the extension without providing it a translator?

Reply:
The only use case is for third-party bundles using translations, but that must be compatible with projects not using translations. But anyway, the main usage is when people wants to use translations, in which case, we need to recommend the implementation.

Reply:
ACK, updated